src/gdb_hostio.c

```diff
@@ -24,24 +24,22 @@
 #include "gdb_hostio.h"
 #include "gdb_packet.h"
 
-int hostio_reply(target_controller_s *tc, char *pbuf, int len)
+int hostio_reply(target_controller_s *const tc, char *const pbuf, const int len)
 {
 	(void)len;
-	int retcode, items, errno_;
-	char c, *p;
-	if (pbuf[1] == '-')
-		p = &pbuf[2];
-	else
-		p = &pbuf[1];
-	items = sscanf(p, "%x,%x,%c", &retcode, &errno_, &c);
-	if (pbuf[1] == '-')
-		retcode = -retcode;
+
+	const bool retcode_is_negative = pbuf[1] == '-';
+
+	unsigned int retcode;
+	unsigned int errno_;
+	char c;
```

Comment:
What is `c` do we know and can we get it a better name?

Reply:
I did not know what it was, had to go through GDB docs
https://sourceware.org/gdb/onlinedocs/gdb/Protocol-Basics.html#Protocol-Basics
hopefully I got it right

Reply:
Looks great, and the addition of the block comment is extremely helpful, thank you!